Automattic/jetpack
# Jetpack: Add block and allow list toggles to WAF settings

### Description
Updates the Jetpack WAF security settings UI to allow the IP block and allow lists to be toggled independently.

## Proposed changes:

* Makes use of the new `jetpack_waf_ip_allow_list_enabled` / `jetpack_waf_ip_block_list_enabled` settings, replacing `jetpack_waf_ip_list`
* Removes the individual manual rules setting and adds independent toggles for each IP list
* Modifies sections dependent on the original setting

### Other information:

## Jetpack product discussion


- https://github.com/Automattic/jetpack-scan-team/issues/1263

## Does this pull request change what data or activity we track or use?


- No

## Testing instructions:






* Checkout this branch
* Start up JT, and enable Jetpack
* Proceed to WAF security settings and toggle the various options in different combinations
* Ensure the settings all perform as expected and that no regressions are introduced
* Upgrade and downgrade and ensure all expected settings are still accessible and functional
* Verify that a blocklisted IP cannot make requests or access the site admin when BFP is enabled
* Ensure that the blocked IP can once again do so once whitelisted
* Ensure cross compatibility exists when Protect is enabled and the settings are toggled from that end
* Ensure backward compatibility exists when the update is applied to an existing site with the prior settings in place

## Screenshots

All Enabled:
![Screen Shot 2024-07-17 at 06 47 41](https://github.com/user-attachments/assets/e8f7f25c-24b2-40fd-9225-31a34f59eff8)

All Disabled:
![Screen Shot 2024-07-17 at 06 46 54](https://github.com/user-attachments/assets/8b2a8aff-513c-4f88-92d6-98d11d0bd693)

WAF Disabled:
![Screen Shot 2024-07-17 at 06 46 02](https://github.com/user-attachments/assets/8523ee2a-30aa-462d-a1d5-26af794fa0ea)

Unsupported:
![Screen Shot 2024-07-17 at 06 45 08](https://github.com/user-attachments/assets/c1861ac1-eb26-4762-bd5d-16092a1563fb)
